### DISCUSSION TITLE
Add (conditional) minimum position

### DISCUSSION THREAD

**Discussion by basbruss** [basbruss/adaptive-cover#225] [Ideas]
Based on the discussion at https://github.com/basbruss/adaptive-cover/issues/171, some users want the cover to have a minimum position.

In my opinion, this conflicts with the integration's main goal:

> Reduce glare on the work area while allowing natural light when possible.

Here's how this conflict arises:
Limiting the cover to only close up to a certain percentage can reduce its effectiveness at minimizing glare. When the sun is low in the sky, the sun's rays hit the window almost horizontally, requiring the cover to close as much as possible to block the glare. However, the sun is only at this low angle for a short period each day, so the times when the cover needs to be almost fully closed are brief.

Some users suggest that a partially open cover can help with ventilation. However, on summer days, the sun reaches a higher angle, meaning the cover can naturally remain more open, allowing for better ventilation. In short, the cover system already allows more ventilation on summer days due to the higher sun elevation.

The two diagrams below illustrate this concept, comparing a winter day and a summer day in the Northern Hemisphere (the same logic applies in the Southern Hemisphere).

Winter (render date `"2024-12-05"`)
![image](https://github.com/basbruss/adaptive-cover/assets/68892092/b2187e2d-a510-4887-9540-257cb9e77096)
Note: the climate settings will set the covers mostly to the default since the glare is not strong enough/present according to the available thresholds. Those are not evaluated in this simulation.

Summer (render date `"2024-06-17"`)
![image](https://github.com/basbruss/adaptive-cover/assets/68892092/5a46598d-2de7-4f60-871f-6818d39c6365)

I'm open to hearing about specific use cases under certain conditions that might benefit from having a minimum cover position at those times.

**Comment by ShooterQ** [comment]
I think adding the minimum position doesn’t mean abandoning your mission statement; just adds better customization for a litany of use cases without hurting the goal of reducing glare, sticking to temp ranges, etc. 

Coming from Adaptive Lighting, I have some lights that flicker and buzz if they are set too low, so I NEED a minimum to keep them working nicely. 

Likewise, I have windows above plants where I’d like them to receive full light no matter what the blind is doing; I could achieve this if I had a minimum or say, 5% or 10%. 

Another example, I want one blind adapting to reduce heat coming into the home, but I always want it open to about 7% minimum because it allows me to look out and see my car. 

As is, I could accomplish this by rezeroing my blind to the desired minimum, but then this takes away my ability to have them close fully in other cases, such as when my security is armed and I want everything covered down to 0%.

**Reply by basbruss** [reply]
Interesting and valid points.

Does this also mean that you do want to allow the cover to cross the minimum when there is no (direct) light on the window, let's say at night or when one of the default values is below the minimum?

**Reply by ShooterQ** [reply]
Yes, I’d think that it’d be another option to give the user in a checkbox, letting them decide. Or by calling it Daytime Minimum. 

I personally disable Adaptive Cover in my security automation when security is armed, and reenable it when it disarmed, which is how I would currently bypass the minimum myself if that option existed but no daytime distinction was available.

**Comment by autumnwalker** [comment]
I have a simple (silly?) use case for minimum position - cat. My cat will attack my cellular shades that cover my patio door if they touch her ears - with the old blueprint I kept the minimum level at 35% and the problem was solved.

**Comment by kineticscreen** [comment]
I have another (silly?) reason for wanting the minimum position - and is why I am currently still using the old Blueprint, but would love to move across to this integration. I'm using the blind cover to stop the sun hitting books on my shelf that is next to it - but it only needs to extend down to 31%, which lines up nicely with the frame of the bottom window. So during the day there is no reason to have the whole window covered, and it looks less intrusive when it lines up with the edge of the window frame.

**Comment by homelab-hobbyist** [comment]
I would like to see a conditional sensor that sets a minimum position based on a window binary sensor. 

If it's open, I want the blind at 60% of its maximum closed percentage. If the window is closed, then I want it closed all the way, if the sun & climate conditions are met.

**Reply by disruptivepatternmaterial** [reply]
Similarly, all my windows have open/close sensors. I would like the shades not to close when the window is open. This worked well using the old blue print, but here I have to write an automation for each window to set it in manual override and then clear that when the window is closed. It is not an ideal situation.

And the minimum position (which I think has been implemented, from what I can tell) is needed for a host of reasons - the on for me is that I have plants in my window and I do not want to shade to close and kill them or knock them off. I also want to let a little light in the room in my office in the heat of the day, so forcing no more closed than 5% is ideal.
